src/pybind/mgr/dashboard/frontend/src/app/shared/services/api-interceptor.service.ts

```diff
@@ -35,14 +35,18 @@ export class ApiInterceptorService implements HttpInterceptor {
 
   intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
     const defaultVersion = '1.0';
+    let version = request.headers.get('version');
```

Comment:
Can you add tests for this?

Reply:
If I understand it correctly, if some http request for e.g. host list wants to set `version=0.1` then I think we also need to add header for it, right? I don't understand here why we want to introduce a new header if we already know for which requests we want to change `Accept` header? As if we follow this approach we will also need to add header in all those requests which requires version change. Why not better just modify the `Accept` header for those requests itself, for e.g. host list request with `version=0.1`
```
list(): Observable<object[]> {
    return this.http.get<object[]>(this.baseURL, { headers: 
      {'Accept': 'application/vnd.ceph.api.v0.1+json'} });
  }
  ```

Reply:
@avanthakkar You're right. @aaSharma14 can ou adapt the code and provide tests?

Reply:
@alfonsomthd In this case we need not change anything in the api-interceptor-service.Currently implemented code can handle everything.

Reply:
@nizamial09 , you can follow the example suggested by @avanthakkar in the above comment to make the version thing work in the front-end too.